### PR TITLE
Initial commit/skeleton for "use mock chassis" story

### PR DESCRIPTION
**Pull request by marios**
OK - so I got the  "I would like to use ironic mock chassis" story. There's an obvious overlap with the stuff that Jay is doing for "As a user I would like to use mock ironic models" - this is just some basic skeleton for now:
- Chassis model  /model/Ironic/chassis.py (just attributes from [api](https://github.com/openstack/ironic/blob/master/doc/source/dev/api-spec-v1.rst#chassis) )
- Ironic driver chassis methods - tuskar/drivers/ironic_chassis_methods.rb 
  As a user I would like to use ironic mock chassis:                       
  
  --> get_all_chassis():  
  --> get_chassis(c_id):  
  --> create_chasis(attrs={}):  
  --> update_chasis(attrs={}): #e.g. add hosts/nodes to the chassis or remove
  --> delete_chasis(c_id)                                                  
- ironic_common.py is probably going to get overwritten by Jay - meant to hold the util methods like 'store'/'retrieve' for the mock objects

Also an overlap with the Rack - obvious ones:
1) user creates a new Rack. All of the hosts in that rack will be placed into a newly created Ironic chassis.
2) user adds a host to a Rack. The chassis corresponding to the rack should be updated to include the new host.
3) user lists a Rack - the chassis attribute is returned in the serialization.
4) user lists all Racks - chassis attribute on all Rack.  
5) user deletes a Rack - delete the associated chassis.                    

marios

### CODE REVIEW

File: tuskar/api/model/Ironic/chassis.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
```
Comment (markmc) [inline tuskar/api/model/Ironic/chassis.py:1]: This is probably the wrong place to put this stuff - it's an implementation detail of the ironic driver, not a part of the tuskar REST API

Comment (markmc) [inline tuskar/api/model/Ironic/chassis.py:1]: Oh, also - avoid the use of upper case package names and you'd need a **init**.py file in this dir for it to be usable

**Comment by markmc** [comment]
Would be interested to see what the high-level tuskar driver API methods that call down to this ironic chassis stuff looks like - e.g. is it a notify_rack_created() method that creates the chassis?

Does tuskar ever need to list all chassis?

**Comment by mfojtik** [comment]
@markmc do you know what is the standard way of doing notifications in OpenStack? Does simple RPC works well or do we need to do some messaging stuff?

**Comment by markmc** [comment]
Don't get too distracted by the "notify" name ... it'd be an in-process hook for the driver to do something when a rack is created, like create the corresponding chassis. Could be called something else

**Comment by markmc** [comment]
Closing this since you said you didn't want it merged yet. I presume you can re-open it when you're ready, but I actually have no idea :)

**Pull request closed without merging**